src/app/data-explorer/querybuilder/querybuilder.component.html

```diff
@@ -0,0 +1,272 @@
+            let addRule = addRule;
+            let addRuleSet = addRuleSet;
+            let removeRuleSet = removeRuleSet
+          "
+        >
+          <button
+            class="add-rule-button"
+            type="button"
+            mat-button
+            color="primary"
+            (click)="addRule()"
+          >
+            <mat-icon>add</mat-icon>
+            Rule
+          </button>
+          <button
+            class="add-rule-button"
+            type="button"
+            mat-button
+            color="primary"
+            *ngIf="addRuleSet"
+            (click)="addRuleSet()"
+          >
+            <mat-icon>add_circle_outline</mat-icon>
+            Rule Set
+          </button>
+          <button
+            type="button"
+            mat-button
+            color="accent"
+            *ngIf="removeRuleSet"
+            (click)="removeRuleSet()"
+          >
+            <mat-icon>remove_circle_outline</mat-icon>
+          </button>
+        </ng-container>
+        <!--  -->
+        <ng-container *queryArrowIcon>
+          <mat-icon ngClass="mat-arrow-icon">chevron_right</mat-icon>
+        </ng-container>
+        <!-- Remove Rule Button -->
+        <ng-container *queryRemoveButton="let rule; let removeRule = removeRule">
+          <button type="button" mat-icon-button color="accent" (click)="removeRule(rule)">
+            <mat-icon>remove</mat-icon>
+          </button>
+        </ng-container>
+        <!--  AND / OR group-->
+        <ng-container *querySwitchGroup="let ruleset; let onChange = onChange">
+          <mat-radio-group
+            *ngIf="ruleset"
+            [(ngModel)]="ruleset.condition"
+            (ngModelChange)="onChange($event)"
+          >
+            <mat-radio-button [style.padding.px]="10" value="and">AND</mat-radio-button>
+            <mat-radio-button [style.padding.px]="10" value="or">OR</mat-radio-button>
+          </mat-radio-group>
+        </ng-container>
+        <!--  -->
+        <ng-container
+          *queryEntity="let rule; let entities = entities; let onChange = onChange"
+        >
+          <mat-form-field
+            appearance="outline"
+            class="query-builder-field mat-form-field-no-padding"
+          >
+            <mat-select
+              [(ngModel)]="rule.entity"
+              (ngModelChange)="onChange($event, rule)"
+            >
+              <mat-option *ngFor="let entity of entities" [value]="entity.value">
+                {{ entity.name }}
+              </mat-option>
+            </mat-select>
+          </mat-form-field>
+        </ng-container>
+        <!-- Field Selector -->
+        <ng-container
+          *queryField="
+            let rule;
+            let fields = fields;
+            let onChange = onChange;
+            let getFields = getFields
+          "
+        >
+          <mat-form-field
+            appearance="outline"
+            class="query-builder-field mat-form-field-no-padding"
```

Comment:
If I removed all stylings related to the Material controls the query builder ends up looking likes this:
![image](https://user-images.githubusercontent.com/110250803/206161894-70088b84-3227-496e-ac69-1558bba52269.png)

Where as with styles it looks like this:
![image](https://user-images.githubusercontent.com/110250803/206162580-3de97361-65cd-4e7a-8677-8e403c7b2c10.png)

We lose a lot of screen real estate due to the height of the controls and the control sizes are terrible. Yes, the alignment of the dropdown icon goes wrong but I think that is preferential to the alternative.

The width of the controls is set in querybuilder.component.scss.
.query-builder-field {
  width: 400px !important;
}

.query-builder-operator {
  width: 160px !important;
}

.query-builder-input {
  width: 300px !important;
}

Reply:
This might be subjective opinion then, but I quite like the non-custom styled controls, so long as there was some margins between each one. Think we need some other opinions 😄 

Could you comment somewhere in the CSS how these controls were overridden? It's the height I couldn't follow how to change. If it was clearer somewhere in comments then someone else would be able to maintain it at least, and it would highlight that these controls have been modified.

Reply:
That does look better actually, proper alignment helps.